src/unix.rs

```diff
@@ -90,7 +90,11 @@ impl Child {
         if self.reaped {
             Ok(())
         } else {
-            self.inner.kill()
+            self.inner.kill()?;
+            let mut status = 0;
+            let id = self.id() as c_int;
+            unsafe { libc::waitpid(id, &mut status, 0) };
```

Comment:
* `waitpid` can be interrupted, so we should at least retry on `EINTR` here similar to what we do in [`try_wait`](https://github.com/alexcrichton/tokio-process/blob/ec5df8afb274202864148c20e8d3400fbdc35b55/src/unix.rs#L119-L139).
* Also, could you put a small comment, for posterity, to explain why we expect `waitpid` to return quickly

Reply:
I think we'll also want to be sure to set `reaped` to `true` to ensure that we don't reuse the pid as well